src/js/brim/entries.js

```diff
@@ -4,20 +4,31 @@ import isEqual from "lodash/isEqual"
 
 type Args = {
   position: number,
-  entries: *[]
+  entries: *[],
+  scrollPositions: *[]
 }
 
-export default function({entries: initEntries, position}: Args) {
+export default function({
+  entries: initEntries,
+  position,
+  scrollPositions: initScrollPositions = []
+}: Args) {
   let entries = [...initEntries]
+  let scrollPositions = [...initScrollPositions]
 
   if (position < -1 || position >= entries.length) {
     throw new Error("Position out of bounds")
   }
 
   return {
-    push(entry: *) {
+    push(entry: *, scrollPos: *) {
       if (!isEqual(entry, this.getCurrentEntry())) {
         entries.splice(position + 1, entries.length, entry)
+        scrollPositions.splice(position + 1, scrollPositions.length, {
+          x: 0,
+          y: 0
+        })
+        scrollPositions[position] = scrollPos
```

Comment:
This is a point that's up for discussion. Technically it's a side effect, however splitting it off into it's own function leads to a lot of extra code.